src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -2450,5 +2450,187 @@ static void M(object? o)
+}
+static class CExt2
+{
+    public static void Add<T>(this C c, T t) => throw null!;
+}
+";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (10,23): error CS0121: The call is ambiguous between the following methods or properties: 'CExt1.Add<T>(C, T)' and 'CExt2.Add<T>(C, T)'
+                //         _ = new C() { o1, Identity(o1 ??= new object()), o1, o2 };
+                Diagnostic(ErrorCode.ERR_AmbigCall, "o1").WithArguments("CExt1.Add<T>(C, T)", "CExt2.Add<T>(C, T)").WithLocation(10, 23),
+                // (10,27): error CS0121: The call is ambiguous between the following methods or properties: 'CExt1.Add<T>(C, T)' and 'CExt2.Add<T>(C, T)'
+                //         _ = new C() { o1, Identity(o1 ??= new object()), o1, o2 };
+                Diagnostic(ErrorCode.ERR_AmbigCall, "Identity(o1 ??= new object())").WithArguments("CExt1.Add<T>(C, T)", "CExt2.Add<T>(C, T)").WithLocation(10, 27),
+                // (10,58): error CS0121: The call is ambiguous between the following methods or properties: 'CExt1.Add<T>(C, T)' and 'CExt2.Add<T>(C, T)'
+                //         _ = new C() { o1, Identity(o1 ??= new object()), o1, o2 };
+                Diagnostic(ErrorCode.ERR_AmbigCall, "o1").WithArguments("CExt1.Add<T>(C, T)", "CExt2.Add<T>(C, T)").WithLocation(10, 58),
+                // (10,62): error CS0121: The call is ambiguous between the following methods or properties: 'CExt1.Add<T>(C, T)' and 'CExt2.Add<T>(C, T)'
+                //         _ = new C() { o1, Identity(o1 ??= new object()), o1, o2 };
+                Diagnostic(ErrorCode.ERR_AmbigCall, "o2").WithArguments("CExt1.Add<T>(C, T)", "CExt2.Add<T>(C, T)").WithLocation(10, 62));
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var collectionInitializer = root.DescendantNodes().OfType<InitializerExpressionSyntax>().Single();
+
+            verifyAnnotation(collectionInitializer.Expressions[0], PublicNullableAnnotation.Annotated);
+            verifyAnnotation(collectionInitializer.Expressions[1], PublicNullableAnnotation.NotAnnotated);
+            verifyAnnotation(collectionInitializer.Expressions[2], PublicNullableAnnotation.NotAnnotated);
+            verifyAnnotation(collectionInitializer.Expressions[3], PublicNullableAnnotation.NotAnnotated);
+
+            void verifyAnnotation(ExpressionSyntax expr, PublicNullableAnnotation expectedAnnotation)
+            {
+                var symbolInfo = model.GetCollectionInitializerSymbolInfo(expr);
+                Assert.Null(symbolInfo.Symbol);
+                foreach (var symbol in symbolInfo.CandidateSymbols)
+                {
+                    Assert.Equal(PublicNullableAnnotation.None, ((IMethodSymbol)symbol).TypeArgumentNullableAnnotations[0]);
+                }
+            }
+        }
```

Comment:
would it be useful to test multiple-element collection initializers? i.e. `_ = new C() { { o1, o2 } }`?

are there any scenarios where reinferring the Add method type arguments would result in different warnings? It seems like it won't affect the type of the containing collection, because we have to specify a type argument for it, so I can't think of any that would really be affected.

Reply:
perhaps a scenario where the same `T` is used for the parameters corresponding to `{ o1, o2 }` but different nullabilities, making sure that a reasonable type argument is inferred.

---
In reply to: [321864825](https://github.com/dotnet/roslyn/pull/38532#discussion_r321864825) [](ancestors = 321864825)

Reply:
Yeah, this is a good test scenario, thanks.

---
In reply to: [321865253](https://github.com/dotnet/roslyn/pull/38532#discussion_r321865253) [](ancestors = 321865253,321864825)